presto-main/src/main/java/com/facebook/presto/metadata/TableHandle.java

```diff
@@ -86,7 +86,7 @@ public boolean equals(Object o)
         return Objects.equals(connectorId, that.connectorId) &&
                 Objects.equals(connectorHandle, that.connectorHandle) &&
                 Objects.equals(transaction, that.transaction) &&
-                Objects.equals(layout, that.layout);
+                Objects.equals(layout.isPresent(), that.layout.isPresent());
```

Comment:
This looks incorrect. For two table handles that have different layouts, now we consider them as the same ? :)